### ISSUE TITLE
Matrix.allocate(pattern) does not work if pattern leaves context

### ISSUE THREAD

**Issue by fuchsto** [dash-project/dash#1]
Use cases like that does not work:

``` c++
matrix_t matrix;
{
   pattern_t pattern = ...
   matrix.allocate(pattern);
}
matrix[0] = 1;
```

This functionality is essential for the hdf5 API.

**Comment by fmoessbauer** [comment]
This problem is even worse: Generally cases where teamspec, sizespec and distspec are specified manually might not work:

```
 dash::NArray<int, 2, long, pattern_t > matrix;

 auto & team = dash::Team::All();
 dash::TeamSpec<2>         ts(team);
 dash::SizeSpec<2>         ss(block_size_x, block_size_y);
 dash::DistributionSpec<2> ds(dash::TILE(1),dash::TILE(1));
  pattern_t pattern(
       ss,
       ds,
       ts,
       team);
 matrix.allocate(pattern);
 auto id = dash::myid();
```

**branch** `bug-matrix-del-alloc`

**Comment by fmoessbauer** [comment]
Debugging showed that this Problem is not related to the matrix, but to the `dash::TilePattern`. If using block pattern, the bug does not occur.

**Comment by fuchsto** [comment]
Fixed defect in copy-construction of TilePattern, waiting for CI.

**Comment by fuchsto** [comment]
Fixed in #40
